docs/operator-manual/declarative-setup.md

```diff
@@ -165,121 +165,126 @@ spec:
     repository URL suffixed with `.git`. Argo CD will **not** follow these redirects, so you have to
     adjust your repository URL to be suffixed with `.git`.
 
-Repository credentials are stored in secrets. Use following steps to configure a repo:
-
-1. Create a secret which contains repository credentials. Consider using [bitnami-labs/sealed-secrets](https://github.com/bitnami-labs/sealed-secrets) to store an encrypted secret
-definition as a Kubernetes manifest.
-2. Register the repository in the `argocd-cm` config map. Each repository must have a `url` field and, depending on whether you connect using HTTPS, SSH, or GitHub App, `usernameSecret` and `passwordSecret` (for HTTPS), `sshPrivateKeySecret` (for SSH), or `githubAppPrivateKeySecret` (for GitHub App).
+Repository details are stored in secrets. To configure a repo, create a secret which contains repository details.
+Consider using [bitnami-labs/sealed-secrets](https://github.com/bitnami-labs/sealed-secrets) to store an encrypted secret definition as a Kubernetes manifest.
+Each repository must have a `url` field and, depending on whether you connect using HTTPS, SSH, or GitHub App, `username` and `password` (for HTTPS), `sshPrivateKey` (for SSH), or `githubAppPrivateKey` (for GitHub App).
 
 Example for HTTPS:
 
 ```yaml
 apiVersion: v1
-kind: ConfigMap
+kind: Secret
 metadata:
-  name: argocd-cm
+  name: private-repo
   namespace: argocd
   labels:
-    app.kubernetes.io/name: argocd-cm
-    app.kubernetes.io/part-of: argocd
-data:
-  repositories: |
-    - url: https://github.com/argoproj/my-private-repository
-      passwordSecret:
-        name: my-secret
-        key: password
-      usernameSecret:
-        name: my-secret
-        key: username
+    argocd.argoproj.io/secret-type: repo-config
```

Comment:
```suggestion
    argocd.argoproj.io/secret-type: repository
```

Should match one of those defined in common.go?

Reply:
Good catch! I forgot to change these after I made the last refactoring

Reply:
I noticed only more thing. @jangraefen can you please delete deprecated repos related fields from [argocd-cm.yaml](https://github.com/argoproj/argo-cd/blob/4d9e8bc2e6da0fa400f5af962b0aa718fa28bb49/docs/operator-manual/argocd-cm.yaml#L72) ?

Reply:
I am already working on a new branch to fix these findings and will create a new PR, once done. Or is there a way to commit on the old branch and have them added to this PR even after it was merged?

Reply:
I think Github does not allow it. Just create new PR please

Reply:
I added a new PR here: https://github.com/argoproj/argo-cd/pull/6540